pkg/controller/direct/gkehub/featuremembership_controller.go

```diff
@@ -176,18 +180,48 @@ func (a *gkeHubAdapter) Delete(ctx context.Context, deleteOp *directbase.DeleteO
 
 func (a *gkeHubAdapter) patchMembershipSpec(ctx context.Context) ([]byte, error) {
 	feature := a.actual
+	mSpecs := feature.MembershipSpecs
+	if mSpecs == nil {
+		mSpecs = make(map[string]featureapi.MembershipFeatureSpec)
+	}
 	// only change the feature configuration for the associated membership
-	feature.MembershipSpecs[a.membershipID] = *a.desired
-	op, err := a.featureClient.Patch(a.featureID, feature).UpdateMask("membershipSpecs").Context(ctx).Do()
+	mSpecs[a.membershipID] = *a.desired
+	feature.MembershipSpecs = mSpecs
+	op, err := a.hubClient.featureClient.Patch(a.featureID, feature).UpdateMask("membershipSpecs").Context(ctx).Do()
```

Comment:
Nit: can we pass an etag or a similar optimistic concurrency mechanism?

Reply:
I created https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/2592 to track the work